src/ast/Event.ts

```diff
@@ -1,27 +1,125 @@
-import DayOfWeek from "./DayOfWeek";
-import Date from "./Date";
-
-class Event extends CalTypes {
-  event: String;
-  calTypes: Array<CalTypes>;
-
-  parse(): void {
-    while (!tokenizer.checkToken(";")) {
-      eventName = tokenizer.getNext();
-      if (tokenizer.checkToken("every") || tokenizer.checkToken("and")) {
-        calTypes.add(new DayOfWeek());
+import Node from "../Node";
+import Tokenizer from "../Tokenizer";
+import { ParserError } from "../errors/ParserError";
+
+export default class Event extends Node {
+  title: string = "";
+  repeating: boolean = false;
+  daysOfWeek: string[] = [];
+  date: string = "";
+  allDay: boolean = false;
+  fromTime: string = "";
+  toTime: string = "";
+  location: string = "";
+  guests: string[] = [];
+
+  parse(tokenizer: Tokenizer): void {
+    // Get what should be the title of the event
+    let currentLine = tokenizer.getLine();
+    let token = tokenizer.pop();
+    if (token === null) {
+      throw new ParserError(`Error on line ${currentLine}: expected an event title`);
+    }
+    this.title = token;
+
+    // Get what should be the date of the event
+    currentLine = tokenizer.getLine();
+    token = tokenizer.pop();
+    if (token === "every") {
```

Comment:
maybe an over-optimization now, but we could put all of our literals into some constants list and reference it as `literals.EVERY` to avoid hardcoding the value here

Reply:
Also yes, all our tokens should probably go into an enum of constants lol